python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -393,3 +395,11 @@ def config_from_pkg_resources(pkg_resource_defs: Sequence[Tuple[str, str]]) -> M
         ) from err
 
     return config_from_yaml_strings(yaml_strings=yaml_strings)
+
+
+T = TypeVar("T")
+
+
+def dedupe_objects_ref_equality(objects: Iterable[T]) -> Sequence[T]:
```

Comment:
Consider:
`dedupe_objects_by_ref`
`dedupe_object_refs`

Reply:
Consider making input optional so it handles `None` so that

`dedupe_objects_ref_equality(sensors) if sensors else []`
becomes
`dedupe_object_refs(sensor)`

Reply:
I get slightly scared by the way we coerce none into lists everywhere but I guess one more won't hurt

Reply:
Just a recommendation. If you don't want to make the core utility function handle `None` (I can see the argument for sure) then you can have a little wrapper thing in `Definitions`

Reply:
Do whatever you feel is appropriate. Was just a light recommendation. I do not feel strongly either way. Meaning if you prefer it as is I am also fine with that.